quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/update/AbstractQuarkusCliUpdateIT.java

```diff
@@ -0,0 +1,138 @@
+import static io.quarkus.test.utils.AwaitilityUtils.untilAsserted;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Properties;
+
+import jakarta.inject.Inject;
+
+import org.apache.http.HttpStatus;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.junit.jupiter.api.Test;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+import io.quarkus.test.bootstrap.QuarkusCliClient;
+import io.quarkus.test.bootstrap.QuarkusCliRestService;
+import io.quarkus.test.logging.Log;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.scenarios.annotations.DisabledOnNative;
+import io.quarkus.test.utils.AwaitilityUtils;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+@QuarkusScenario
+@DisabledOnNative // Only for JVM verification
+public abstract class AbstractQuarkusCliUpdateIT {
+    @Inject
+    static QuarkusCliClient cliClient;
+
+    protected final DefaultArtifactVersion oldVersion;
+    protected final DefaultArtifactVersion newVersion;
+
+    public AbstractQuarkusCliUpdateIT(DefaultArtifactVersion oldVersion, DefaultArtifactVersion newVersion) {
+        this.oldVersion = oldVersion;
+        this.newVersion = newVersion;
+    }
+
+    /**
+     * Perform basic update to new stream and check version in pom.xml matches
+     */
+    @Test
+    public void versionUpdateTest() throws ParserConfigurationException, IOException, SAXException {
+        QuarkusCliRestService app = createAppBeforeUpdate();
+
+        updateAppToNewStream(app);
+
+        DefaultArtifactVersion updatedVersion = getQuarkusAppVersion(app);
+        assertEquals(newVersion.getMajorVersion(), updatedVersion.getMajorVersion(),
+                "Major version for app updated to " + newVersion + "should be " + newVersion.getMajorVersion());
+        assertEquals(newVersion.getMinorVersion(), updatedVersion.getMinorVersion(),
+                "Minor version for app updated to " + newVersion + " should be " + newVersion.getMinorVersion());
+
+        Log.info("Starting updated app");
+        // start the updated app and verify that basic /hello endpoint works
+        app.start();
+        untilAsserted(() -> app.given().get("/hello").then().statusCode(HttpStatus.SC_OK),
+                AwaitilityUtils.AwaitilitySettings.usingTimeout(Duration.ofSeconds(10))
+                        .timeoutMessage("Updated app failed to expose working /hello endpoint"));
+    }
+
+    protected void checkPropertiesUpdate(Properties oldProperties, Properties expectedNewProperties) throws IOException {
```

Comment:
this task will repeat therefore I'd prefer to have it in the FW.
If you know that you:

- have app
- run update
- expect properties to be xyz

then it should be one method call with "app" and "expected property after", "expected property missing" etc. all depends on common task

Reply:
I'm not sure if this should be shifted into the framework. Do we expect to test the updates elsewhere?
I don't like the idea of splitting the test logic into two projects. And this method seems to me to be tightly integrated with the test logic.
Also having this somehow complicated method (it needs, to create app, update it - all with versions and app's seteup, run asserts, ...) in a framework would be IMHO very impractical if we want to change the methods behaviour or ad hoc disable failing assert or something in the test suite.

Reply:
well, I obviously disagree, but you are at Helm, follow your guts. if it should be in TS, let it be in TS. please choose different reviewer as I am OOO.

Reply:
cc @rsvoboda

Reply:
Hey. +1 to have this in framework / [QuarkusCLIUtils.java](https://github.com/quarkus-qe/quarkus-test-framework/blob/main/quarkus-test-cli/src/main/java/io/quarkus/test/util/QuarkusCLIUtils.java)

We are not the only consumers of test framework, Camel Quarkus QE team is using it too.

Reply:
PR for these test methods into framework - https://github.com/quarkus-qe/quarkus-test-framework/pull/1239

Reply:
methods moved to framework.